perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2718,6 +2720,208 @@ public void updateSponsorshipValidity(PerunSession sess, Member sponsoredMember,
 		return membersManagerImpl.getSponsorshipsExpiringInRange(sess, from, to);
 	}
 
+	@Override
+	public void moveMembership(PerunSession sess, Vo vo, User sourceUser, User targetUser) throws MemberNotExistsException, AlreadyMemberException, ExtendMembershipException {
+		Member sourceMember = this.getMemberByUserId(sess, vo, sourceUser.getId());
+		List<Group> directGroups = getPerunBl().getGroupsManagerBl().getMemberDirectGroups(sess, sourceMember);
+
+		List<Group> synchronizedGroups = new ArrayList<>();
+		try {
+			for (Group group : directGroups) {
+				Attribute attrSynchronizeEnabled = getPerunBl().getAttributesManagerBl().getAttribute(sess, group, getPerunBl().getGroupsManager().GROUPSYNCHROENABLED_ATTRNAME);
+				if ("true".equals(attrSynchronizeEnabled.getValue())) {
+					synchronizedGroups.add(group);
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			throw new InternalErrorException(e);
+		}
+
+		Member targetMember;
+		try {
+			// create targetMember and add him to all direct groups that are not synchronized
+			directGroups.removeAll(synchronizedGroups);
+			targetMember = this.createMember(sess, vo, targetUser, directGroups);
```

Comment:
Is it possible that targetUser already is a member of this vo?

Reply:
We don't want to merge two existing memberships for now, therefore AlreadyMemberException is thrown in such a case.